Examples/Scripts/FreeSurferPipelineBatch-long.sh

```diff
@@ -0,0 +1,138 @@
+ExcludeVisits=(HCA6002237_V1_MR HCA6002238_V1_MR)
+#Longitudinal template labels, one per each subject.
+Templates=(HCA6002236_V1_V2 HCA6002237_V1_V2 HCA6002238_V1_V2)
+
+EnvironmentScript="<HCPInstallDir>/scripts/SetUpHCPPipeline.sh" #Pipeline environment script
+
+# Requirements for this script
+#  installed versions of: FSL, FreeSurfer, Connectome Workbench (wb_command), gradunwarp (HCP version)
+#  environment: HCPPIPEDIR, FSLDIR, FREESURFER_HOME, CARET7DIR, PATH for gradient_unwarp.py
+
+# If you want to use FreeSurfer 5.3, change the ${queuing_command} line below to use
+# ${HCPPIPEDIR}/FreeSurfer/FreeSurferPipeline-v5.3.0-HCP.sh
+
+#Set up pipeline environment variables and software
+source "$EnvironmentScript"
+
+# Log the originating call
+echo "$@"
+
+#NOTE: syntax for QUEUE has changed compared to earlier pipeline releases,
+#DO NOT include "-q " at the beginning
+#default to no queue, implying run local
+QUEUE="long.q"
+#QUEUE="hcp_priority.q"
+
+#parallel options
+parallel_mode=BUILTIN
+max_jobs=4
+
+#TEMPLATE stage must be run before TIMEPOINTS stage
+start_stage=TEMPLATE
+end_stage=TIMEPOINTS
+
+########################################## INPUTS ########################################## 
+#Scripts called by this script do assume they run on the outputs of the PreFreeSurfer Pipeline
+######################################### DO WORK ##########################################
+
+for i in ${!Subjects[@]}; do
+  Subject=${Subjects[i]}
+  #Subject's time point list, @ separated.  
+  TPlist=(`identify_timepoints $Subject`)
+  #Array with timepoints
+  IFS=, read -ra Timepoints <<< "$TPlist"
+  #Freesurfer longitudinal average template label
+  LongitudinalTemplate=${Templates[i]}
+
+  #Longitudinal FreeSurfer Input Variables
+  SubjectID="$Subject" #FreeSurfer Subject ID Name
+  SubjectDIR="${StudyFolder}/${Subject}/T1w" #Location to Put FreeSurfer Subject's Folder
+  T1wImage="${StudyFolder}/${Subject}/T1w/T1w_acpc_dc_restore.nii.gz" #T1w FreeSurfer Input (Full Resolution)
+  T1wImageBrain="${StudyFolder}/${Subject}/T1w/T1w_acpc_dc_restore_brain.nii.gz" #T1w FreeSurfer Input (Full Resolution)
+  T2wImage="${StudyFolder}/${Subject}/T1w/T2w_acpc_dc_restore.nii.gz" #T2w FreeSurfer Input (Full Resolution)
+
+  if [ "$parallel_mode" != FSLSUB ]; then #fsl_sub does not allow nested submissions
+  if [[ "${command_line_specified_run_local}" == "TRUE" || "$QUEUE" == "" ]] ; then
+      echo "About to locally run ${HCPPIPEDIR}/FreeSurfer/LongitudinalFreeSurferPipeline.sh"
+      #NOTE: fsl_sub without -q runs locally and captures output in files
+      queuing_command=("$FSLDIR/bin/fsl_sub")
+  else
+      echo "About to use fsl_sub to queue ${HCPPIPEDIR}/FreeSurfer/LongitudinalFreeSurferPipeline.sh"
+      queuing_command=("$FSLDIR/bin/fsl_sub" -q "$QUEUE")
+  fi
+  else
+  	queuing_command=()
+  fi
+
+  #DO NOT PUT timepoint-specific options here!!!
+  cmd=("${queuing_command[@]}" "$HCPPIPEDIR"/FreeSurfer/LongitudinalFreeSurferPipeline.sh \
```

Comment:
```suggestion
  cmd=(${queuing_command[@]+"${queuing_command[@]}"} "$HCPPIPEDIR"/FreeSurfer/LongitudinalFreeSurferPipeline.sh \
```
Tolerate empty array in bash < 5.0, in case we use `set -u`.

Reply:
ok